packages/es-components/src/components/controls/buttons/DropdownButton.js

```diff
@@ -40,25 +45,120 @@ const StyledButtonLink = styled(Button)`
   }
 `;
 
+const TAB_KEY_CODE = 9;
+const UP_ARROW_CODE = 38;
+const DOWN_ARROW_CODE = 40;
+
+function getFocusables(node) {
+  const focusableElements = node.querySelectorAll('button');
+  const firstFocusable = focusableElements[0];
+  const lastFocusable = focusableElements[focusableElements.length - 1];
+
+  return {
+    focusableElements,
+    firstFocusable,
+    lastFocusable
+  };
+}
+
+function isCurrentlyActive(node) {
+  return document.activeElement === node;
+}
+
+function focusTrap(node) {
+  const { firstFocusable, lastFocusable } = getFocusables(node);
+
+  function handleTabFocus(event) {
```

Comment:
Could this function (and isCurrentlyActive) be useful for other components, and if so should we move it into the utils folder? Just throwing that out there, it might not be feasible.

Reply:
Yeah, it's possible. I didn't extract it out because it isn't fully fleshed out for all potential focusable element types. I would be OK doing it in a separate PR in the future if the need arises.